src/api/Events.ts

```diff
@@ -3,12 +3,14 @@
  * Copyright FINOS FDC3 contributors - see NOTICE file
  */
 
-/** Type representing a handler function for events from the Desktop Agent. 
- * @param {FDC3Event} event The handler function will be passed an `FDC3Event` Object 
- * providing details of the event (such as a change of channel membership for the app) as the only 
- * parameter.
+/** Type representing a handler function for events from the Desktop Agent
+ * or a PrivateChannel. 
+ * @param event The handler function will be passed an `FDC3Event` or
+ * `PrivateChannelEvent` Object providing details of the event (such as a
+ * change of channel membership for the app, or type of context listener added)
+ * as the only parameter.
  */
-export type EventHandler = (event: FDC3Event) => void;
+export type EventHandler = (event: FDC3Event | PrivateChannelEvent ) => void;
```

Comment:
Is there a good common base type or marker interface recommendation for this in other languages?

Reply:
Ah the dreaded union type again.

The event types are just strings really, we're just using the enums to govern the set as we do with errors - they are all that really differs. The type names are mutually exclusive between the fdc3 interface events and privateChannel interface events, but it felt like they should be different enums.

I haven't included a common anscestor for the two event types (`FDC3Event` & `PrivateChannelEvent`), which is I guess what you are after! I can go add that and have both extend it with specific enums for the types. That would mean EventHandler could use that ancestor here instead...

I'll go take a look at this - I suspect typescript enums are going to make the above more difficult than it should be... Might have to switch to string unions...

Reply:
Checkout https://github.com/finos/FDC3/commit/c47b7b2c38c40444360870f6c684d00524982cd2 
Hopefully it'll alleviate your issue here by introducing a base ApiEvent type that the others restrict down to specific messages.